agent/containers/images/image-test-script.sh

```diff
@@ -0,0 +1,283 @@
+
+wait_keypress 120
+
+# If somebody wants to write their own pbench-tool-meister-start, great, then
+# that person can change the assumption on where the persistent volume is for
+# the Tool Data Sink, and how the "sysinfo", "init", and "end" commands
+# operate with respect to data storage.
+
+export pbench_run=/var/tmp/test-run-dir
+rm -rf ${pbench_run}
+mkdir ${pbench_run}
+printf -- "${pbench_run}\n" > ${pbench_run}/.path
+export pbench_log=${pbench_run}/pbench.log
+
+printf -- "\n\nThe benchmark run directory will be found inside:\n\n"
+
+find ${pbench_run} -ls
+
+wait_keypress 120
+
+
+# Register tools
+
+printf -- "\nNow, let's pick hosts to collect data from, as well as which tools we wish to register for that host.\n\n"
+
+wait_keypress 120
+
+while :; do
+    read -p "Hostname (or press 'ENTER' for default '$CONTROLLER'): "
+    hostvar=${REPLY:-${CONTROLLER}}
+    if ping -c 1 -W 1 ${hostvar} 2>&1 1>/dev/null
+    then
+        echo $hostvar >> ${pbench_run}/remotes.lis
+        read -p "Another host? (y/n)" anotherhost
+        if [ $anotherhost != "y" ]
+        then
+            break
+        fi
+    else
+        printf -- "Host specified either does not exist or is unreachable, please try again:\n"
+    fi
+done
+printf -- "\nDone selecting hosts:\n"
+cat ${pbench_run}/remotes.lis
+printf -- "\nNow to select tools.\n\n"
+
+# Register the specified tools, will be recorded in
+# ${pbench_run}/tools-v1-default
+while :; do
+    read -p "Tool name (type 'help' to see options): " tool
+    if [ -z "$tool" ]
+    then
+        printf -- "Tool name empty. Please write a tool name.\n"
+    else
+        if [ $tool == "help" ]
+        then
+            printf -- "\nAvailable tools:\n"
+            python3 -c "import sys, json; meta = json.load(open(sys.argv[1])); print('  Transient:', *[f'\t{tool}' for tool in meta['transient'].keys()], '  Persistent:', *[f'\t{tool}' for tool in meta['persistent'].keys()], sep='\n')" /opt/pbench-agent/tool-scripts/meta.json
+        else
+            cmd="pbench-register-tool --name=${tool} --remotes=@${pbench_run}/remotes.lis"
+            printf -- "\n$cmd\n"
+            $cmd
+            read -p "Another tool? (y/n)" anothertool
+            if [ $anothertool != "y" ]
+            then
+                break
+            fi
+        fi
+    fi
+done
+printf -- "\nDone registering tools.\n\n"
+ls -lR ${pbench_run}/tools-v1-default
+
+printf -- "\n\n"
+wait_keypress 120
+
+
+# Start Tool Meister containers on remote host
+
+printf -- "\n\nNow please start the Tool Meister container on the specified host...\n\n\t$ podman run --name pbench-agent-tool-meister \\ \n\t\t--network host --ulimit nofile=65536:65536 --rm -d \\ \n\t\t-e REDIS_HOST=${REDIS_HOST} \\ \n\t\t-e REDIS_PORT=${REDIS_PORT} \\ \n\t\t-e PARAM_KEY=tm-default-\$(hostname -f) \\ \n\t\t-e _PBENCH_TOOL_MEISTER_LOG_LEVEL=debug \\ \n\t\tquay.io/pbench/pbench-agent-tool-meister-${DISTRO}:${TAG}\n\n"
```

Comment:
I don't think we win any prizes for performing this in a single line -- just put it in double quotes and lay it out the way you want it to appear:
```
printf -- "

Now please start the Tool Meister container on the specified host...
        \$ podman run --name pbench-agent-tool-meister \\
                --network host --ulimit nofile=65536:65536 --rm -d \\
                -e REDIS_HOST=${REDIS_HOST} \\
                -e REDIS_PORT=${REDIS_PORT} \\
                -e PARAM_KEY=tm-default-\$(hostname -f) \\
                -e _PBENCH_TOOL_MEISTER_LOG_LEVEL=debug \\
                quay.io/pbench/pbench-agent-tool-meister-${DISTRO}:${TAG}

"
```
This could also be done using the idiom that you used on line 186.